app/js/services/QueryBuilderService.js

```diff
@@ -59,25 +65,27 @@ function QueryBuilderService(
         genes = genes.map(gene => gene["entrezgene"]);
 
         // Submit our query
-        queryResource.submit({ diseases, genes }).$promise.then(
-          res => {
-            NotificationService.notify({
-              type: "success",
-              message: `<span class="material-icons" aria-hidden="true">check_circle</span> Classifier #${
-                res.id
-              } submitted!`
-            });
-            $rootScope.$emit('MODAL_CLOSED');
-            $rootScope.$emit('QUERY_SUBMITTED');
-          },
-          error => {
-            console.log(error);
-            NotificationService.notify({
-              type: "error",
-              message: `Failed to submit classifier. Error: ${error}`
-            });
-          }
-        );
+        queryResource
+          .submit({ diseases, genes, subscribe: isMailchimpOptIn })
+          .$promise.then(
+            res => {
+              NotificationService.notify({
+                type: "success",
+                message: `<span class="material-icons" aria-hidden="true">check_circle</span> Classifier #${
+                  res.id
+                } submitted!`
+              });
+              $rootScope.$emit("MODAL_CLOSED");
+              $rootScope.$emit("QUERY_SUBMITTED");
+            },
+            error => {
+              console.log(error);
+              NotificationService.notify({
+                type: "error",
+                message: `Failed to submit classifier. Error: ${error}`
+              });
+            }
+          );
```

Comment:
This code all looks good. Thank you, as usual, for cleaning up the formatting and making things a little more readable!

I do have one question. It seems like an odd choice to collect the `isMailchimpOptIn` during the `submitClassifier()` call to the back-end here rather than at the time the e-mail address is collected initially (below, via the `USER_ENDPOINT` call at line 109-117). Is there a reason it was done this way around?

Reply:
Hmm, good point. I don't think there was a particular reason it was done this way, but when Kurt and I paired on the backend, it seemed like it made most sense to modify the classifier endpoint.

Reply:
Ok. It's not a show stopper, but I thought I would ask. It's probably not worth changing now, but maybe if the API has to be revisited in the future then that would be the time.